pallets/metaverse/src/lib.rs

```diff
@@ -244,21 +349,187 @@ pub mod pallet {
 
 			ensure!(metaverse_info.is_frozen, Error::<T>::OnlyFrozenMetaverseCanBeDestroyed);
 
-			MetaverseOwner::<T>::remove(&metaverse_id, metaverse_info.owner);
+			MetaverseOwner::<T>::remove(metaverse_info.owner, &metaverse_id);
 			Metaverses::<T>::remove(&metaverse_id);
 			Self::deposit_event(Event::<T>::MetaverseDestroyed(metaverse_id));
 			Ok(().into())
 		}
+
+		/// Register metaverse for staking
+		/// only metaverse owner can register for staking
+		#[pallet::weight(10_000)]
+		pub fn register_metaverse(origin: OriginFor<T>, metaverse_id: MetaverseId) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			let metaverse_info = Self::get_metaverse(metaverse_id).ok_or(Error::<T>::MetaverseInfoNotFound)?;
+			ensure!(metaverse_info.owner == who, Error::<T>::NoPermission);
+
+			ensure!(
+				!RegisteredMetaverse::<T>::contains_key(&metaverse_id),
+				Error::<T>::AlreadyRegisteredForStaking
+			);
+
+			T::Currency::reserve(&who, T::MetaverseRegistrationDeposit::get())?;
+
+			RegisteredMetaverse::<T>::insert(metaverse_id.clone(), who.clone());
+
+			Self::deposit_event(Event::<T>::NewMetaverseRegisteredForStaking(metaverse_id, who));
+
+			Ok(().into())
+		}
+
+		/// Lock up and stake balance of the origin account.
+		/// New stake will be applied at the beginning of the next round.
+		#[pallet::weight(100_000)]
+		pub fn stake(
```

Comment:
In estate, we used to have bond_more and bond_less. In metaverse, we only have stake. Does it mean the value is the total value the origin wanted to stake? Then we need to increase or decrease the staked amount in staking info?

Reply:
Yes, staker just needs to call stake extrinsic to stake the amount they want to stake into the metaverse. This will increase the staked amount of the current round.